interface/forms/questionnaire_assessments/questionnaire_assessments.php

```diff
@@ -0,0 +1,191 @@
+?>
+<html>
+<head>
+    <title id="main_title"><?php echo xlt('Questionnaire'); ?></title>
+    <?php Header::setupHeader([]); ?>
+    <link href="<?php echo $GLOBALS['assets_static_relative']; ?>/lforms/webcomponent/styles.css" media="screen" rel="stylesheet" />
+    <script>
+        function saveQR() {
+            top.restoreSession();
+            let qr = LForms.Util.getFormFHIRData('QuestionnaireResponse', 'R4');
+            let formElement = document.getElementById("formContainer");
+            let data = LForms.Util.getUserData(formElement, true, true, true);
+            document.getElementById('lform_response').value = JSON.stringify(data);
+            document.getElementById('questionnaire_response').value = JSON.stringify(qr);
+            if (!document.getElementById('questionnaire').value) {
+                let lForm = JSON.parse(document.getElementById('lform').value);
+                let qFhir = LForms.Util.getFormFHIRData("Questionnaire", 'R4', data);
+                document.getElementById('questionnaire').value = JSON.stringify(qFhir);
+            }
+            return true;
+        }
+
+        function initUpdate() {
+            // Merge QuestionnaireResponse
+            let qResponse = JSON.parse(document.getElementById('questionnaire_response').value);
+            let lForm = JSON.parse(document.getElementById('lform').value);
+            let responseData = LForms.Util.mergeFHIRDataIntoLForms("QuestionnaireResponse", qResponse, lForm, "R4");
+            $(".isNew").toggleClass('d-none');
+            LForms.Util.addFormToPage(responseData, 'formContainer');
+        }
+
+        function initNewForm() {
+            let lform = <?php echo js_escape($lform); ?>;
+            let qFhir = <?php echo js_escape($q_json); ?>;
+            let formName = <?php echo js_escape($questionnaire_form); ?>;
+            let data;
+            if (lform) {
+                data = JSON.parse(lform);
+            } else if (qFhir) {
+                let qData = JSON.parse(qFhir);
+                data = LForms.Util.convertFHIRQuestionnaireToLForms(qData, 'R4');
+            } else {
+                alert(xl('Error Missing Form.'));
+                parent.closeTab(window.name, false);
+            }
+            document.getElementById('questionnaire').value = JSON.stringify(qFhir);
+            document.getElementById('lform').value = JSON.stringify(data);
+            document.getElementById('form_name').value = jsAttr(formName);
+            document.getElementById('code_type').value = jsAttr('LOINC');
+            document.getElementById('code').value = jsAttr(data.code);
+            if (typeof data.copyrightNotice !== 'undefined' && data.copyrightNotice > '') {
+                document.getElementById('copyright').value = jsAttr(data.copyrightNotice);
+                document.getElementById('copyrightNotice').innerHTML = jsText(data.copyrightNotice);
```

Comment:
agree when inserting html via innerHTML that is needs to be html escaped, however when setting `.value`'s above, pretty sure do not need to escape that.
(honestly, though, I don't have a 100% grasp on when need to escape js for html)

Reply:
lol aw come on now!

Reply:
I think I agree that normally an escape wouldn't be needed but when I look to escape I look at the source of the data. If from outside of my control say from web as in this case, I tend to be leery and escape. Although, we know I'm not know as an escaping king by no means.
I went back and forth on this and tested both ways then decided to err on being safe. If you feel strongly I can remove.